core/runtime/src/main/java/io/quarkus/devservices/crossclassloader/runtime/RunningDevServicesRegistry.java

```diff
@@ -53,6 +55,31 @@ public void closeAllRunningServices(String launchMode) {
         servicesIndexedByLaunchMode.remove(launchMode);
     }
 
+    public void closeRemainingRunningServices(UUID uuid, String launchMode, Collection<DevServiceOwner> ownersToKeep) {
+        Set<RunningService> services = servicesIndexedByLaunchMode.get(launchMode);
+        var iterator = servicesIndexedByConfig.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<ComparableDevServicesConfig, RunningService> entry = iterator.next();
+            DevServiceOwner owner = entry.getKey().owner();
+            UUID serviceAppUuid = entry.getKey().applicationInstanceId();
+            if (owner.launchMode().equals(launchMode) && Objects.equals(serviceAppUuid, uuid)
+                    && !ownersToKeep.contains(owner)) {
+                iterator.remove();
+                RunningService serviceToClose = entry.getValue();
+                services.remove(serviceToClose);
+                try {
+                    System.out.println("Closing dev service for " + owner.featureName() + " in launch mode " + launchMode + " "
+                            + serviceToClose.containerId());
+                    serviceToClose.close();
+                } catch (Exception e) {
+                    // We don't want to fail the shutdown hook if a service fails to close
+                    e.printStackTrace();
+                }
```

Comment:
I wish we had a better pattern for this. I've had to do similar things in the test classloading rewrite, where JUnit would swallow exceptions that users really needed to see. Here we want to swallow it, just not totally silently. But a `printStackTrace()` feels like a minor smell.